error-column matching

FrostKiwi opened the pull request:
Thanks to @antaalt and his https://github.com/KhronosGroup/glslang/pull/3614, we now have the error column reported by `glslangValidator`. This Pull Request adds compatibility for this.

| Before this PR (or without `--error-column`) | This PR |
|------------------------------------|---------|
| ![image](https://github.com/user-attachments/assets/149eb10b-7d77-44f9-9af2-eb3886285ead) | ![image](https://github.com/user-attachments/assets/a1ac81be-a2f6-408b-a370-2ee737c0cb03) |

# Functionality
If you have updated `glslangValidator` to a version that supports this, you can specify `"--error-column` among `"glsllint.glslangValidatorArgs`. Like this: `"glsllint.glslangValidatorArgs": ["--error-column"]`

Since there is no official release version yet, nightly builds of `glslangValidator` can be had here: https://github.com/KhronosGroup/glslang/releases/tag/main-tot

If `--error-column` is among the arguments, a new regex is used to pull the column and display the error squiggly accordingly.

antaalt commented:
Nice work, I had some code in local to test it, it could be even better to use an optional regex group for the column tag in order for the regex to support both log format without having to care if glslang support the option, such as `/(WARNING|ERROR):\s+(\d|.*):(\d+):(?:(\d+):)\s+(.*)/`, this way, no need to check for the option support, and it will work for both versions of glslang

FrostKiwi commented:
> it could be even better to use an optional regex group for the column tag in order for the regex to support both log format without having to care if glslang support the option, such as `/(WARNING|ERROR):\s+(\d|.*):(\d+):(?:(\d+):)\s+(.*)/`

Didn't consider this. A good change, I added it and removed the check for the `--error-column` argument. I had to modify the regex some more, because it kept matching against `stdin` in the non-`--error-column` case. Code got definitely simpler 👍

hsimpson commented:
@FrostKiwi thanks for the PR, I will have a look into it.
